test/e2e_node/device_plugin_test.go

```diff
@@ -589,6 +589,81 @@ func testDevicePlugin(f *framework.Framework, pluginSockDir string) {
 				framework.Fail(fmt.Sprintf("stale device assignment after pod deletion while kubelet was down allocated=%v error=%v", allocated, err))
 			}
 		})
+
+		ginkgo.It("Can schedule a pod with a restartable init container [NodeAlphaFeature:SidecarContainers]", func(ctx context.Context) {
+			podRECMD := fmt.Sprintf("devs=$(ls /tmp/ | egrep '^Dev-[0-9]+$') && echo stub devices: $devs && sleep %s", sleepIntervalWithRestart)
+			pod1 := e2epod.NewPodClient(f).CreateSync(ctx, makeBusyboxPodWithRestartableInitContainer(SampleDeviceResourceName, podRECMD))
+			deviceIDRE := "stub devices: (Dev-[0-9]+)"
+			devID1, err := parseLog(ctx, f, pod1.Name, pod1.Spec.InitContainers[0].Name, deviceIDRE)
+			framework.ExpectNoError(err, "getting logs for pod %q/%q", pod1.Name, pod1.Spec.InitContainers[0].Name)
+			gomega.Expect(devID1).To(gomega.Not(gomega.Equal("")), "pod1's restartable init container requested a device but started successfully without")
+
+			devID2, err := parseLog(ctx, f, pod1.Name, pod1.Spec.Containers[0].Name, deviceIDRE)
+			framework.ExpectNoError(err, "getting logs for pod %q/%q", pod1.Name, pod1.Spec.Containers[0].Name)
+			gomega.Expect(devID2).To(gomega.Not(gomega.Equal("")), "pod1's regular container requested a device but started successfully without")
+
+			gomega.Expect(devID1).NotTo(gomega.Equal(devID2), "pod1's restartable init container and regular container should not share the same device")
+
+			v1alphaPodResources, err = getV1alpha1NodeDevices(ctx)
+			framework.ExpectNoError(err)
+
+			v1PodResources, err = getV1NodeDevices(ctx)
+			framework.ExpectNoError(err)
+
+			framework.Logf("v1alphaPodResources.PodResources:%+v\n", v1alphaPodResources.PodResources)
```

Comment:
please don't use the `v1alpha1` podresources API in new code unless there is a strong reason to